encodings/dict/src/compress.rs

```diff
@@ -62,12 +107,15 @@ pub trait DictEncoder {
 /// Dictionary encode primitive array with given PType.
 /// Null values in the original array are encoded in the dictionary.
 pub struct PrimitiveDictBuilder<T> {
-    lookup: HashMap<Value<T>, u64>,
+    lookup: HashMap<private::Value<T>, u64, FxBuildHasher>,
```

Comment:
AFK right now, but did we use FoldHash here or the default std one?

Reply:
we'd use w/e hashbrown was using which upon inspection seems to be foldhash

Reply:
i am by no means a hashing expert, but after reading several blog posts i saw a lot praise for rustc-hash, and it did  provide a measurable difference (though the biggest difference was definitely from getting rid of ToBytes)

Reply:
I think for primitive T you could just get rid of value and make NativePType hash? Yes right now we use foldhash

Reply:
you want it for floats as well

Reply:
hence the need for value so you can define hash+eq

Reply:
see my new doc comment on Value ;)

Reply:
forgot about the floats... ignore me